main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -493,20 +517,80 @@ DataTableView.prototype._showRows = function(start, onDone) {
   }, this._sorting);
 };
 
+DataTableView.prototype._onChangeGotoPage = function(elmt, evt) {
+  var gotoPageNumber = parseInt($('input#viewpanel-paging-current-input').val());
+  
+  if(typeof gotoPageNumber != "number" || isNaN(gotoPageNumber) || gotoPageNumber == "") { 
+    $('input#viewpanel-paging-current-input').val(this._currentPageNumber); 
+    return;
+  }
+  
+  if(gotoPageNumber > this._lastPageNumber) gotoPageNumber = this._lastPageNumber;
+  if(gotoPageNumber < 1) gotoPageNumber = 1;
+  
+  this._currentPageNumber = gotoPageNumber;
+  this._showRows((gotoPageNumber - 1) * this._pageSize);
+};
+
+DataTableView.prototype._onKeyDownGotoPage = function(elmt, evt) {
+  var keyDownCode = event.which;
+  
+  if([38, 40].indexOf(keyDownCode) == -1) return;
+  
+  evt.preventDefault();
+
+  var newPageValue = $('input#viewpanel-paging-current-input')[0].value;
+  if(keyDownCode == 38) {  // Up arrow
+    if(newPageValue <= 1) return;
+    this._onClickPreviousPage(elmt, evt);
+  }
+    
+  if(keyDownCode == 40) {  // Down arrow
+    if(newPageValue >= this._lastPageNumber) return;
+    this._onClickNextPage(elmt, evt);
+  }
+  
+  var rehookDelay = 50;
+  var rehookCurrentPageInput = function() { 
```

Comment:
Why do we need to rehook anything? Are you working around a specific browser bug? It would be good to add some comments to explain what is going on here.

Reply:
> Why do we need to rehook anything? Are you working around a specific browser bug? It would be good to add some comments to explain what is going on here.

Here, the code logic is simple. The <input> field, from what I understand, gets deleted with a page change. The data gets requeried for each page change, so a new field is created « sometimes » after. The joy with client/server roundtrip is that is take a very different time for the exact same task… network contribution and it's overhead I guess…

So, I don't know « when » the new field will exists. So, I come back quickly, but if the new field doesn't exist, then, I'll just comeback latter, as long as it takes…

It is not the best coding choice, it's the less disruptive code change choice. Less change, potentially less breaking, but maybe not. It's not a hack, but it's not pure. A refactoring of the UI would be better, but development path wise, this now, and the Vue.js change after that, then fixing this would be very easy, I think. More than now anyway.

Reply:
I have heard you talking about "the Vue.js change" before but I am not aware of anyone planning to do that soon, unless @lisa761 has expressed the wish to do this during the GSoC project?

I understand you do not want to introduce code changes that are too involved but I am not comfortable with merging a hack like that, it could easily break if the update takes longer then 50ms.

Reply:
@wetneb But is it an actual "break" that's not quickly recoverable by a user? Where the UX is completely locked up?  I don't think so, right?  This would just appear to the user that things don't update...they click in another area of the UI to refocus in some other area, then the UX is recovered.  No?  Or am I missing something?

Reply:
> I have heard you talking about "the Vue.js change" before but I am not aware of anyone planning to do that soon, unless @lisa761 has expressed the wish to do this during the GSoC project?

I was referring to #1312. But it seams it's dropped since a few days…

> I understand you do not want to introduce code changes that are too involved but I am not comfortable with merging a hack like that, it could easily break if the update takes longer then 50ms.

This is not what it does: if after 50ms, it doesn't see the <input> it reschedule a check in the next 50 ms. And it will do that as long as it takes.

What I don't like about this, is that if there is a recurring error, then the mechanism will always be retriggered. That it's limitation. Now, that « shouldn't » happen, but it could in particular circonstances.

Like, if this feature was to be pushed in v3.4, I wouldn't not put « Infinite Paging » in it. But, in v3.5, I would put it.

Regards,
   Antoine

Reply:
Oups. I'm sorry. My comment is about code not pushed on server. 

Now, what I have is clean. It's recursive, and has a fail safe to stop after _x_ times.

Regards,
   Antoine